### PR TITLE
fix: Improve Terraform agent<->resource association testing

### PR DESCRIPTION
**Pull request by kylecarbs**
An agent could be associated with multiple resources prior to this, which broke all workspace access (and doesn't make any sense). This adds tests based on Terraform output. The raw Terraform statefiles are pushed for fast testing, and can be iterated on by editing the Terraform and running `generate.sh`.

This should improve test times and provide a much more stable solution for resource association!

### CODE REVIEW

**Comment by bpmct** [comment]
Does this encompass the bug with relations in #1884?

**Comment by kylecarbs** [comment]
@bpmct yup!

**Review by f0ssel — APPROVED** [review]
go code lgtm, can't speak to the tf files

File: provisioner/terraform/testdata/generate.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+for d in */; do
```
Comment (deansheather) [inline provisioner/terraform/testdata/generate.sh:6]: I appreciate that you've included the script for this :)

**Review by deansheather — APPROVED** [review]
The test cases are very epic

**Pull request merged**